File: VB/Example/Customer.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Linq
Imports System.Runtime.CompilerServices

Namespace Example
	Public Class Customer
		Implements INotifyPropertyChanged
		Private firstName_Renamed As String
		Private id_Renamed As Integer
		Private parentID_Renamed As Integer

		Public Sub New()
			FirstName = "FirstName"
			ParentID = 0
			id_Renamed = 0
		End Sub

		Public Sub New(ByVal firstName As String, ByVal parentID As Integer, ByVal id As Integer)
			Me.FirstName = firstName
			Me.ParentID = parentID
			Me.id_Renamed = id
		End Sub

		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Private Sub NotifyPropertyChanged(<CallerMemberName> Optional ByVal propertyName As String = "")
            If PropertyChangedEvent IsNot Nothing Then
                PropertyChangedEvent.Invoke(Me, New PropertyChangedEventArgs(propertyName))
            End If
        End Sub

        Public Property FirstName() As String
			Get
				Return firstName_Renamed
			End Get
			Set(ByVal value As String)
				If value <> firstName_Renamed Then
					firstName_Renamed = value
					NotifyPropertyChanged()
				End If
			End Set
		End Property
		Public ReadOnly Property ID() As Integer
			Get
				Return id_Renamed
			End Get
		End Property
		Public Property ParentID() As Integer
			Get
				Return parentID_Renamed
			End Get
			Set(ByVal value As Integer)
				If value <> parentID_Renamed Then
					parentID_Renamed = value
					NotifyPropertyChanged()
				End If
			End Set
		End Property
	End Class
End Namespace

File: CS/Example/Customer.cs
using System;
using System.ComponentModel;
using System.Linq;
using System.Runtime.CompilerServices;

namespace Example
{
    public class Customer : INotifyPropertyChanged
    {
        string firstName;
        int id;
        int parentID;

        public Customer()
        {
            FirstName = "FirstName";
            ParentID = 0;
            id = 0;
        }

        public Customer(string firstName, int parentID, int id)
        {
            FirstName = firstName;
            ParentID = parentID;
            this.id = id;
        }

        public event PropertyChangedEventHandler PropertyChanged;

        private void NotifyPropertyChanged([CallerMemberName] string propertyName = "")
        {
            if(PropertyChanged != null)
                PropertyChanged.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

        public string FirstName {
            get { return firstName; }
            set {
                if(value != firstName)
                {
                    firstName = value;
                    NotifyPropertyChanged();
                }
            }
        }
        public int ID {
            get { return id; }
        }
        public int ParentID {
            get { return parentID; }
            set {
                if(value != parentID)
                {
                    parentID = value;
                    NotifyPropertyChanged();
                }
            }
        }
    }
}